### PR TITLE
A better revised version for M2N2

### PR DESCRIPTION
**Pull request by zhangsunny**
I have made some changes based on the original M2N2 code, as follows:

- ​​remove​ unused code in M2N2.py for better readability.

- modify the code to align with the original paper and official implementation, including:
    - M2N2 generates anomaly scores for each timestep in a window, whereas the previous implementation used only the score at the last timestep per window, that is inconsistent to the official settings.
    - The previous version allocated a validation set but omitted the validation process (likely an oversight). Now, I have added corresponding code to support both validation-enabled and validation-free setting.
    - The original testing setup used a stride of 1 and averaged overlapping-window regions to match anomaly score length with the input sequence. While this post-processing improved metric results (significantly), it violated online inference constraints by utilizing future data. To strictly follow the original paper's methodology, I implement a right-side-padding for anomaly scores, although this differs from the authors' approach of applying padding directly to the input sequences.
    - add an `infer_mode` option to support both offline and online inference modes.

- refactor the `run_M2N2` function to support richer parameter configurations, and give more reasonable parameters, especially the latent dimension must be smaller than the input one to enforce the bottleneck structure in the encoder-decoder architecture, as this compression mechanism is essential for anomaly detection.

### CODE REVIEW

**Comment by qhliu26** [comment]
Thank you so much for your contribution! I will merge the commits into the main branch once we are done with the test. A quick question, does the new implementation support custom stride as well (https://github.com/TheDatumOrg/TSB-AD/blob/2d507e1e544fc24f662c4641a8af83a03b2193fa/TSB_AD/models/M2N2.py#L590)?

**Comment by zhangsunny** [comment]
Unfortunately not. Since the official implementation (https://github.com/carrtesy/M2N2/blob/97f8ef2343e18b13518ff92c6d61a9811fd01679/Exp/MLP.py#L208) uses the non-overlapping window setting, the current version doesn't support a custom stride.

If needed, please keep the previous code both at (https://github.com/TheDatumOrg/TSB-AD/blob/2d507e1e544fc24f662c4641a8af83a03b2193fa/TSB_AD/models/M2N2.py#L469) and (https://github.com/TheDatumOrg/TSB-AD/blob/2d507e1e544fc24f662c4641a8af83a03b2193fa/TSB_AD/models/M2N2.py#L590).

By the way, using the maximum `As.append(torch.max(A.clone().detach(), dim=-1)[0])` or the mean value `As.append(torch.mean(A.clone().detach(), dim=-1))` might be more appropriate. The former has better performance during testing.
